service/src/main/java/bio/terra/profile/service/azure/AzureService.java

```diff
@@ -54,6 +61,26 @@ public List<AzureManagedAppModel> getAuthorizedManagedAppDeployments(
         .toList();
   }
 
+  /**
+   * Gets the resource providers in a subscription that are in either the "Registered" or
+   * "Registering" state.
+   *
+   * @param tenantId Azure tenant ID associated with the given subscription.
+   * @param subscriptionId Azure subscription ID to be checked for providers
+   * @return Set of registered or registering resource providers.
+   */
+  public Set<String> getRegisteredProvidersForSubscription(UUID tenantId, UUID subscriptionId) {
+    var resourceManager = crlService.getResourceManager(tenantId, subscriptionId);
+    var providers = resourceManager.providers();
+    return providers.list().stream()
+        .filter(
+            provider ->
+                provider.registrationState().equals("Registered")
+                    || provider.registrationState().equals("Registering"))
+        .map(Provider::namespace)
```

Comment:
What does line 80 do? Convert from a `Provider` to just the string `namespace`? I'm wondering if we should return the full `Provider` instead.

Reply:
Or update the documentation to make it clear that it returns the provider namespaces.

Reply:
Returning the namespace should be sufficient unless there is a compelling need for consumers of this method to have the full provider. I'll update the doc.